semperfiwebdesign/all-in-one-seo-pack
# Refactor get_robots_meta()

Issue #1798

## Proposed changes

Fixes the following notice "undefined variable: post_type in aioseop_class.php". This happens on the category archive page when Posts are set to NOFOLLOW globally and categories are set to NOINDEX globally.

## Types of changes

- Bugfix (non-breaking change which fixes an issue)

## Checklist


## Testing instructions

First, in order to reproduce, do these steps:

1. Go to the General Settings, make sure Categories are NOINDEX and enable NOFOLLOW for Posts.
2. Create a Post and assign it to a category or leave it uncategorized.
3. View the relevant category page and you'll see the notice.*

*(WP_DEBUG has to be enabled of course).

Then install this PR, make sure that the notice went away and that this didn't break anything else.
You might want to play around with custom posts and taxonomies.
